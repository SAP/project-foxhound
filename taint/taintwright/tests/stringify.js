function test() {
    // Ensure location.hash is non-empty
    location.hash = "initial";

    // Source
    let str = location.hash;

    // Create obj to stringify
    let obj = {
        prefix: "test",
        s: str
    }

    // Propagation
    let out = JSON.stringify(obj);

    // Sink
    location.hash = out;
}
