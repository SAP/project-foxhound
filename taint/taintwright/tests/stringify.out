[
    {
        "begin": 22,
        "end": 30,
        "flow": [
            {"operation": "function"},
            {"operation": "location.hash"},
            {"operation": "JSON.stringify"},
            {"operation": "location.hash"}
        ]
    }
]